nmdc_automation/import_automation/activity_mapper.py

```diff
@@ -52,140 +51,169 @@ def __init__(
-                    )
-
-                    filemeta = os.stat(updated_file)
-
-                    md5 = get_md5(updated_file)
-
-                    dobj = self.runtime.minter(self.data_object_type)
-
-                    self.nmdc_db.data_object_set.append(
-                        nmdc.DataObject(
-                            file_size_bytes=filemeta.st_size,
-                            name=file_destination_name,
-                            url=f"{self.url}/{self.nucelotide_sequencing_id}/{workflow_execution_id}/{file_destination_name}",
-                            data_object_type=data_object_dict["data_object_type"],
-                            type=self.data_object_type,
-                            id=dobj,
-                            md5_checksum=md5,
-                            description=data_object_dict["description"].replace(
-                                "{id}", self.nucelotide_sequencing_id
-                            ),
+                    # sequencing_dir = os.path.join(self.root_dir, self.nucelotide_sequencing_id)
+                    try:
+                        os.makedirs(self.root_dir)
+                    except FileExistsError:
+                        logger.debug(f"{self.root_dir} already exists")
+
+                    export_file = os.path.join(self.root_dir, file_destination_name)
+
+                    try:
+                        os.link(import_file, export_file)
+                    except FileExistsError:
+                        logger.debug(f"{export_file} already exists")
+
+                    filemeta = os.stat(export_file)
+                    md5 = get_md5(export_file)
+                    data_object_id = self.runtime.minter(self.data_object_type)
+                    do_record = {
+                        "id": data_object_id,
+                        "type": self.data_object_type,
+                        "name": file_destination_name,
+                        "url": f"{self.url}/{self.nucelotide_sequencing_id}/{file_destination_name}",
+                        "file_size_bytes": filemeta.st_size,
+                        "md5_checksum": md5,
+                        "data_object_type": data_object_dict["data_object_type"],
+                        "description": data_object_dict["description"].replace(
+                            "{id}", self.nucelotide_sequencing_id
                         )
-                    )
-                    self.objects[data_object_dict["data_object_type"]] = (
-                        data_object_dict["input_to"],
-                        [data_object_dict["output_of"]],
-                        dobj,
-                    )
+                    }
+                    db.data_object_set.append(DataObject(**do_record))
+                    has_output.append(data_object_id)
+        update = {
+            "collection": "data_generation_set",
+            "filter": {"id": self.nucelotide_sequencing_id},
+            "update": {"has_output": has_output}
```

Comment:
probably need some logic to make sure we aren't clobbering an existing value

Reply:
updated the query to "$push" to the has_output array which will not clobber